pkg/kademlia/kademlia.go

```diff
@@ -145,14 +145,15 @@ func (k *Kademlia) DumpNodes(ctx context.Context) (_ []*pb.Node, err error) {
 func (k *Kademlia) Bootstrap(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 	defer k.bootstrapFinished.Release()
+	defer k.timeTrack(time.Now(), "Bootstrap")
 
 	if !k.lookups.Start() {
 		return context.Canceled
 	}
 	defer k.lookups.Done()
 
 	if len(k.bootstrapNodes) == 0 {
-		k.log.Warn("No bootstrap address specified.")
+		k.log.Warn("No bootsrap address specified", zap.Stringer(k.routingTable.self.Type.String(), k.routingTable.self.Id))
```

Comment:
No need to log `self`.

Reply:
Also no need to call `String()` when using `zap.Stringer`. This is the sole purpose of `zap.Stringer` -- to avoid calling `String()` prematurely.